src/main/java/com/github/monster860/fastdmm/FastDMM.java

```diff
@@ -805,4 +870,59 @@ private void loop() {
 			Display.sync(60);
 		}
 	}
+
+	private String getAppData() {
+		String workingDirectory;
+		//here, we assign the name of the OS, according to Java, to a variable...
+		String OS = (System.getProperty("os.name")).toUpperCase();
+		//to determine what the workingDirectory is.
+		//if it is some version of Windows
+		if (OS.contains("WIN"))
+		{
+			//it is simply the location of the "AppData" folder
+			workingDirectory = System.getenv("AppData");
+            workingDirectory += "\\FastDMM\\";
+		}
+		//Otherwise, we assume Linux or Mac
+		else
+		{
+			//in either case, we would start in the user's home directory
+			workingDirectory = System.getProperty("user.home");
+			//if we are on a Mac, we are not done, we look for "Application Support"
+			workingDirectory += "/Library/Application Support";
```

Comment:
As I said in the OP, I tested this on a Linux VM and it worked fine. How it worked is beyond me, though.